homeassistant/components/rpi_camera/camera.py

```diff
@@ -5,82 +5,44 @@
 import subprocess
 from tempfile import NamedTemporaryFile
 
-import voluptuous as vol
-
-from homeassistant.components.camera import PLATFORM_SCHEMA, Camera
+from homeassistant.components.camera import Camera
 from homeassistant.const import CONF_FILE_PATH, CONF_NAME, EVENT_HOMEASSISTANT_STOP
-from homeassistant.helpers import config_validation as cv
-
-_LOGGER = logging.getLogger(__name__)
 
-CONF_HORIZONTAL_FLIP = "horizontal_flip"
-CONF_IMAGE_HEIGHT = "image_height"
-CONF_IMAGE_QUALITY = "image_quality"
-CONF_IMAGE_ROTATION = "image_rotation"
-CONF_IMAGE_WIDTH = "image_width"
-CONF_TIMELAPSE = "timelapse"
-CONF_VERTICAL_FLIP = "vertical_flip"
-
-DEFAULT_HORIZONTAL_FLIP = 0
-DEFAULT_IMAGE_HEIGHT = 480
-DEFAULT_IMAGE_QUALITY = 7
-DEFAULT_IMAGE_ROTATION = 0
-DEFAULT_IMAGE_WIDTH = 640
-DEFAULT_NAME = "Raspberry Pi Camera"
-DEFAULT_TIMELAPSE = 1000
-DEFAULT_VERTICAL_FLIP = 0
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_FILE_PATH): cv.isfile,
-        vol.Optional(CONF_HORIZONTAL_FLIP, default=DEFAULT_HORIZONTAL_FLIP): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=1)
-        ),
-        vol.Optional(CONF_IMAGE_HEIGHT, default=DEFAULT_IMAGE_HEIGHT): vol.Coerce(int),
-        vol.Optional(CONF_IMAGE_QUALITY, default=DEFAULT_IMAGE_QUALITY): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=100)
-        ),
-        vol.Optional(CONF_IMAGE_ROTATION, default=DEFAULT_IMAGE_ROTATION): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=359)
-        ),
-        vol.Optional(CONF_IMAGE_WIDTH, default=DEFAULT_IMAGE_WIDTH): vol.Coerce(int),
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_TIMELAPSE, default=1000): vol.Coerce(int),
-        vol.Optional(CONF_VERTICAL_FLIP, default=DEFAULT_VERTICAL_FLIP): vol.All(
-            vol.Coerce(int), vol.Range(min=0, max=1)
-        ),
-    }
+from .const import (
+    CONF_HORIZONTAL_FLIP,
+    CONF_IMAGE_HEIGHT,
+    CONF_IMAGE_QUALITY,
+    CONF_IMAGE_ROTATION,
+    CONF_IMAGE_WIDTH,
+    CONF_OVERLAY_METADATA,
+    CONF_OVERLAY_TIMESTAMP,
+    CONF_TIMELAPSE,
+    CONF_VERTICAL_FLIP,
+    DOMAIN,
 )
 
+_LOGGER = logging.getLogger(__name__)
 
 def kill_raspistill(*args):
     """Kill any previously running raspistill process.."""
+    _LOGGER.debug("killall raspistill")
     subprocess.Popen(
         ["killall", "raspistill"], stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT
     )
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Raspberry Camera."""
+    _LOGGER.debug("setup_platform")
```

Comment:
Please add a guard clause here that checks if discovery_info is None and return is so.

Reply:
I would ; but could you point me to documentation for `setup_platform` method ?
Because I didn't find it and I don't precisely understand what arguments we receive in there (for instance I had a lot of problem before I could understand that received `config` is empty, and I still don't understand why...)
Here specifically, I would like to understand what value is received in `discovery_info` and in which case.
thank you

Reply:
`discovery_info` here will be the fourth argument that we passed in `load_platform`.

We need the guard clause to gaurd if users add a platform yaml config section by mistake. That would also set up the platform and we don't want that.

Reply:
Here's the relevant docs:
https://developers.home-assistant.io/docs/creating_component_generic_discovery#loading-platforms-when-configured-via-configurationyaml